crt_portal/api/views.py

```diff
@@ -284,3 +285,36 @@ def get(self, request):
             return HttpResponse(status=400)
         except IndexError:
             return HttpResponse(status=500)
+
+
+class ReferralResponse(APIView):
+    """
+    API endpoint that enables referral letter previews and email send status
+
+
+    Example: api/referral-response/
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        report_id = request.data['report_id']
+        report = get_object_or_404(Report, pk=report_id)
+        template_id = request.data['template_id']
+        template = get_object_or_404(ResponseTemplate, pk=template_id)
+        action = request.data['action']
+        preview = {'preview': {}}
+        if not action:
+            return Response({'response': '500: No action provided', **preview})
```

Comment:
I don't think this actually returns a 500

also - 500 should be in the event of a unexpected server error (our code broke) this should probably be 400

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors

https://docs.djangoproject.com/en/4.2/ref/request-response/#jsonresponse-objects

```suggestion
            return JsonResponse(
              {
                'message': 'No action provided',
                **preview,
              }, status=400
              )
```